express-api/src/services/properties/propertiesServices.ts

```diff
@@ -0,0 +1,42 @@
+import { AppDataSource } from '@/appDataSource';
+import { Building } from '@/typeorm/Entities/Building';
+import { Parcel } from '@/typeorm/Entities/Parcel';
+
+const propertiesFuzzySearch = async (keyword: string, limit?: number) => {
+  const parcels = await AppDataSource.getRepository(Parcel)
+    .createQueryBuilder('parcel')
+    .leftJoinAndSelect('parcel.Agency', 'agency')
+    .leftJoinAndSelect('parcel.AdministrativeArea', 'adminArea')
+    .leftJoinAndSelect('parcel.Evaluations', 'evaluations')
+    .leftJoinAndSelect('parcel.Fiscals', 'fiscals')
+    .where(`parcel.pid::text like '%${keyword}%'`)
+    .orWhere(`parcel.pin::text like '%${keyword}%'`)
+    .orWhere(`agency.name like '%${keyword}%'`)
+    .orWhere(`adminArea.name like '%${keyword}%'`)
+    .orWhere(`parcel.address1 like '%${keyword}%'`)
+    .take(limit)
+    .getMany();
```

Comment:
If you wanted to do this without the query builder, this seems to be working for me:
``` ts
 const parcels = await AppDataSource.getRepository(Parcel).find({
    relations: {
      Agency: true,
      AdministrativeArea: true,
      Evaluations: true,
      Fiscals: true,
    },
    where: [
      {
        Agency: {
          Name: Like(`%${keyword}%`),
        },
      },
      {
        AdministrativeArea: {
          Name: Like(`%${keyword}%`),
        },
      },
      {
        Address1: Like(`%${keyword}%`),
      },
      !isNaN(parseInt(keyword)) ? { PID: parseInt(keyword) } : undefined,
      !isNaN(parseInt(keyword)) ? { PIN: parseInt(keyword) } : undefined,
    ],
  });
```

Reply:
This should result in PID and PIN only working for exact rather than partial matches though right?

Reply:
Hmm, seems to be. Should PIDs and PINs partially match? 
Also, I think this is case-specific. There's ILike provided by TypeORM as well, which is case-insensitive.

Reply:
I dunno, I assumed they should but I guess you could argue they should be exact. Maybe a Lindsay question.